packages/interfaces/src/service/navigation-service.interface.ts

```diff
@@ -33,8 +33,17 @@ export interface Navigator {
      * {@link EventDispatcherService} will emit a {@link NavigationEvent}
      *
      * @param {string} path path to the location
+     * @param query query string parameters to append to the URL
      */
-    navigate(path: string): void;
+    navigate(path: string, query?: Location['query']): void;
+
+    /**
+     * Update the current location with new data
+     *
+     * @emits navigate - When the location has been changed the
+     * {@link EventDispatcherService} will emit a {@link NavigationEvent}
+     */
+    navigate(params: Partial<Location>): void;
```

Comment:
Perhaps name this `location`?
```suggestion
    navigate(location: Partial<Location>): void;
```

Reply:
ok if I rename it to `to`? 🙂

Reply:
never mind, I renamed to `location` like you suggested 🙂